pkg/controller/cheinstallation/cheinstallation_controller_test.go

```diff
@@ -475,6 +482,49 @@ func TestReconcile(t *testing.T) {
 			AssertThatCheCluster(t, cheCluster.Namespace, cheCluster.Name, cl).DoesNotExist()
 			AssertThatCheInstallation(t, cheInstallation.Namespace, cheInstallation.Name, cl).HasConditions(InstallationFailed("checlusters.org.eclipse.che \"codeready-workspaces\" not found"))
 		})
+
+		t.Run("should delete CheCluster when deleting CheInstallation", func(t *testing.T) {
+			// given
+			cheInstallation := NewInstallation()
+			deletionTS := metav1.NewTime(time.Now())
+			cheInstallation.SetDeletionTimestamp(&deletionTS) // mark resource as deleted
+			cheOperatorNS := cheInstallation.Spec.CheOperatorSpec.Namespace
+			cheCluster := NewCheCluster(cheOperatorNS)
+			_, r := configureClient(t, cheInstallation,
+				newCheNamespace(cheOperatorNS, v1.NamespaceActive),
+				NewOperatorGroup(cheOperatorNS),
+				NewSubscription(cheOperatorNS),
+				cheCluster)
+			request := newReconcileRequest(cheInstallation)
+
+			// when
+			_, err := r.Reconcile(request)
+
+			// then
+			require.NoError(t, err)
+			// expect CheCluster to be not found
+			updatedCheCluster := orgv1.CheCluster{}
+			err = r.client.Get(context.TODO(), types.NamespacedName{
+				Namespace: cheCluster.Namespace,
+				Name:      cheCluster.Name,
+			}, &updatedCheCluster)
+			require.Error(t, err)
+			assert.True(t, apierrors.IsNotFound(err))
+
+			// assume another reconcile loop happens when the CheCluster is deleted for good
+			request = newReconcileRequest(cheInstallation)
+			// when
+			_, err = r.Reconcile(request)
+			// then
+			require.NoError(t, err)
+			// in that case, expect CheInstallation to have no finalizers
+			updatedCheInstallation := v1alpha1.CheInstallation{}
+			err = r.client.Get(context.TODO(), types.NamespacedName{
+				Name: cheInstallation.Name,
+			}, &updatedCheInstallation)
+			require.NoError(t, err)
+			assert.Empty(t, updatedCheInstallation.GetFinalizers())
```

Comment:
just a detail - wouldn't it be easier to add a new method `HasNoFinalizer` to the `CheInstallationAssertion` type? and call the assertion here?

Reply:
good idea! Done f537f76